src/m365/tenant/commands/auditlog/auditlog-report.ts

```diff
@@ -0,0 +1,261 @@
+
+  private getBatchedPromises(AuditContentLists: AuditContentList[], batchSize: number): void {
+    for (let i: number = 0; i < AuditContentLists.length; i += batchSize) {
+      const PromiseRequestBatch : Promise<AuditlogReport[]>[]= AuditContentLists.slice(i, i + batchSize < AuditContentLists.length ? i + batchSize : AuditContentLists.length).map((AuditContentList : AuditContentList) => {
+        return this.getAuditLogReportforSingleContentURL(AuditContentList.contentUri);
+      })
+      this.batchedPromises.push(PromiseRequestBatch);
+    }
+  }
+
+  private getBatchedAuditlogData(logger: Logger, batchNumber: number, resolve: () => void, reject: (err: any) => void): void {
+    
+    if (this.verbose) {
+      logger.logToStderr(`Starting Batch : ${batchNumber}`);
+    }
+    Promise
+      .all(this.batchedPromises[batchNumber])
+      .then((data : any) => {
+        this.completeAuditReports.push(data);
+        if (batchNumber < this.batchedPromises.length - 1) {
+          this.getBatchedAuditlogData(logger, ++batchNumber, resolve, reject)
+        }
+        else {
+          resolve();
+        }
+      }, err => { return Promise.reject(err) });
+  }
+
+  private getAuditLogReportforSingleContentURL(auditURL: string): Promise<AuditlogReport[]> {
+    const requestOptions: any = {
+      url: auditURL,
+      headers: {
+        accept: 'application/json;'
+      },
+      responseType: 'json'
+    };
+
+    return request
+      .get<AuditlogReport[]>(requestOptions);
+  }
+  
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '-c, --contentType <contentType>',
+        description: 'Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP'
+      },
+      {
+        option: '-s, --startTime [startTime]',
+        description: 'Start time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      },
+      {
+        option: '-e, --endTime [endTime]',
+        description: 'End time of logs to be retrieved. Start time and end time must both be specified (or both omitted) and must be less than or equal to 24 hours apart.'
+      }
+    ];
+
+    const parentOptions: CommandOption[] = super.options();
+    return options.concat(parentOptions);
+  }
+
+  public validate(args: CommandArgs): boolean | string {
+    if ((<any>AuditContentTypes)[args.options.contentType] === undefined) {
+      return `${args.options.contentType} is not a valid value for the contentType option. Allowed values are ${Object.keys(AuditContentTypes).join(' | ')}`;
+    }
+
+    if (args.options.startTime || args.options.endTime) {
+      if ((args.options.startTime && !(args.options.endTime)) || (!(args.options.startTime) && args.options.endTime)) {
+        return `startTime and endTime must both be specified (or both omitted)`;
+      }
```

Comment:
Could we simplify this? If we omit the endtime and provide a starttime, just add 24h to the endtime.

This might be useful if we search back 5 days for example. I do not see any reason of specifying the enddate if I just want to see what happened 5 days ago

Reply:
That is a brilliant suggestion. I was being a robot developer and was blindly following the specs without thinking from the user's shoes. Your suggestion is amazing and definitely increases the usability for the users sticking on to the recommended specs.

Another question If I may @plamber , how about a case where the user enters, `endTime `alone? _just thinking out wild here_. Do you think we can generate `startTime `automatically which is 24 hours before the entered `endTime`? Or do you think it is not quite needed and don't see that situation coming into picture?

Reply:
Not sure about this use case. I am seeing it more with the startTime variable.

Reply:
Thanks @plamber for the suggestion. Will stick on with your approach in that case. We can always update it in future if similar use case comes up.

Another question, do you think we need to update the [specs ](https://github.com/pnp/cli-microsoft365/issues/1739#issue-665769955)since it is mentioned that `Start time and end time must both be specified (or both omitted)` for `startTime `and `endTime `parameters.

Sorry for bombarding with questions..

Reply:
Yes, please update the specs too

Reply:
Thanks @plamber for the confirmation. Guess I do not have permission to update the specs from my end. 
![image](https://user-images.githubusercontent.com/10369964/103514126-05892d80-4e92-11eb-8838-59fbf3360c1a.png)

Do you think you will be able to update it from your end? Thanks for all the guidance.

Reply:
@arjunumenon: just send me the exact wording and I will update the specs

Reply:
> @arjunumenon: just send me the exact wording and I will update the specs

Hello @plamber  - Below is command specs which I thought could be given. Please free to update the specs if needed from your end if that makes more sense. I would be totally fine with that. 🙂

Option | Description
-- | --
-c, --contentType <contentType> | Audit content type of logs to be retrieved, should be one of the following: AzureActiveDirectory, Exchange, SharePoint, General, DLP.
-s, --startTime [startTime] | Start time of logs to be retrieved. Start time and end time must be less than or equal to 24 hours apart. Start time is mandatory if End time is specified.
-e, --endTime [endTime] | End time of logs to be retrieved. Start time and end time must be less than or equal to 24 hours apart. If End time is not specified, command will assume the End time to be 24 hours after the Start time.
-o, --output [output] | Output type. json,text. Default text
--verbose | Runs command with verbose logging
--debug | Runs command with debug logging

Reply:
Updated the spects